libraries/chain_kv/include/b1/session/rocks_session.hpp

```diff
@@ -399,6 +405,25 @@ inline rocksdb::ColumnFamilyHandle* session<rocksdb_t>::column_family_() const {
    return nullptr;
 }
 
+template <typename Iterator_traits>
+session<rocksdb_t>::rocks_iterator<Iterator_traits>::rocks_iterator(const rocks_iterator& it)
+    : m_session{ it.m_session } {
+   if (!m_session) {
+      return;
+   }
+
+   auto new_iterator = rocks_iterator<iterator_traits>{};
```

Comment:
I prefer this style of initialization.  There is very consistent style here where the type is always inferred by what is on the left side.  Since auto is allowed for use within this code base, I feel there is less clutter and it is easier to read through the code.

Reply:
Resolved by using a delegating constructor.